UVicMartletplace/martletplace
# Gavin/114 create api endpoint stubs for search engine

# Description
This PR includes the stub methods/endpoints to implement for the search engine API. 

Additionally, it includes some definitions of BaseModel and Enum classes which help to clearly define the allowed inputs and (successful response) outputs.


Closes #114 

## How to Test
1. Run 'docker-compose up --build' in the root of the martletplace repo.
2. Enter the following into your browser:
http://localhost:8221/api/search?authorization=123abc&query=refrigerators&latitude=34.2331&longitude=-124.2323&page=1&limit=20&status=AVAILABLE&searchType=LISTINGS&sort=RELEVANCE
- Your output should look like this:
![image](https://github.com/UVicMartletplace/martletplace/assets/90437591/2af3faf4-c197-404e-8afa-e2f5e868a98d)

3. Enter the following into your browser:
http://localhost:8221/docs
- The page should look like this:
![image](https://github.com/UVicMartletplace/martletplace/assets/90437591/8d411ede-8610-4a4f-9d23-004cc14e140b)
4. For each endpoint shown, click 'try it out', fill out the parameter fields, and execute. Check the outputs.

## Checklist
- [x] The code includes tests if relevant (not yet relevant)
